astropy/table/tests/test_operations.py

```diff
@@ -1540,6 +1540,20 @@ def test_vstack_structured_column(self):
             "               --   four",
         ]
 
+    def test_vstack_inputs_not_modified(self):
+        """Tests that inputs are not modified, see issue #16119"""
+        t1 = Table(data=dict(x=[1, 2, 3], y=["a", "b", "c"]))
+
+        rows = list(t1)  # Table -> list of Rows
+        assert isinstance(rows[0], table.Row)
```

Comment:
I'd make this a bit stronger:
```
rows0 = rows[0]
table.vstack(rows)
assert rows[0] is rows0
```
I don't think you need the rest of the test (or write it for a `Quantity`, but really fine without too).

Reply:
Good idea; I strengthened the test with the `assert rows[0] is rows0` check.  I think it is still useful to do the null test that Tables remain Tables too, so I left that (while also following the `tables[0] is tables0` pattern).  I didn't add `Quantity` testing, since I'm not familiar with stacking those into a Table.